### PR TITLE
Apply sprite frames editor FPS value correctly upon _animation_select

### LINKED ISSUE #26979: FPS value in AnimatedSprite (2D) applied to wrong animation when switching to another
**Godot version:**
3.1-RC3

**OS/device including version:**
Windows 10

**Issue description:**
If you have 2 animations in a AnimatedSprite (2D) with the same FPS value, and you try to change the FPS of one of them, and then you click in the second one, then the FPS change is wrongly applied to second one.

If the animations do not have the same FPS value them the change is omitted/discarded and both animations preserve the original value.

**Steps to reproduce:**
1. Create a AnimatedSprited (2D).
2. Assign a new SpriteFrames.
3. Click on the SpriteFrames to open the panel.
4. Create 2 animations.
5. Click in the first one.
6. Change the FPS value. Do not press enter key.
7. Click in the other animation.

**Minimal reproduction project:**
[Test.zip](https://github.com/godotengine/godot/files/2958059/Test.zip)

### PR DESCRIPTION
**Pull request by dozingpip**
Closes #26979

### CODE REVIEW

**Comment by freestraws** [comment]
The minus didn't show up in my diff when I look at the pull request so I'm not sure how to fix the test it didn't pass?

File: editor/plugins/sprite_frames_editor_plugin.cpp [hunk 1]
```diff
@@ -456,6 +456,10 @@ void SpriteFramesEditor::_animation_select() {
 	if (updating)
 		return;
 
+	double value = anim_speed->get_line_edit()->get_text().to_double();
+	if (value != frames->get_animation_speed(edited_anim))
```
Comment (akien-mga) [inline editor/plugins/sprite_frames_editor_plugin.cpp]: I would use `!Math::is_equal_approx(value, rames->get_animation_speed(edited_anim))` instead as the current check is likely to fail most of the time due to precision issues.

**Comment by akien-mga** [comment]
Could you squash commits together?

**Comment by freestraws** [comment]
I'm pretty sure I did the squash correctly?

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**